src/pymovements/utils/archives.py

```diff
@@ -149,18 +158,35 @@ def _extract_tar(
         Path to the directory the file will be extracted to.
     compression: str | None
         Compression filename suffix.
+    resume: bool
+        Resume if archive was already previous extracted.
+    verbose: int
+        Print messages for resuming each dataset resource.
     """
     with tarfile.open(source_path, f'r:{compression[1:]}' if compression else 'r') as archive:
-        if sys.version_info < (3, 12):  # pragma: <3.12 cover
-            archive.extractall(destination_path)
-        else:  # pragma: >=3.12 cover
-            archive.extractall(destination_path, filter='tar')
+        for member in tqdm(archive.getmembers()):
+            if resume:
+                member_dest_path = os.path.join(destination_path, member.name)
+                if (
+                        os.path.exists(member_dest_path) and
+                        member.name[-4:] in _ARCHIVE_EXTRACTORS and
+                        member.size == os.path.getsize(member_dest_path)
+                ):
+                    if verbose:
+                        print(f'Skipping {member.name} due to previous extraction')
+                    continue
+            if sys.version_info < (3, 12):  # pragma: <3.12 cover
+                archive.extract(member.name, destination_path)
+            else:  # pragma: >=3.12 cover
+                archive.extract(member.name, destination_path, filter='tar')
 
 
 def _extract_zip(
         source_path: Path,
         destination_path: Path,
         compression: str | None,
+        resume: bool,
```

Comment:
can you add a `*,` before `resume`? This is PR a breaking change either way, so we can make it more future proof.